tiledb/api/c_api/current_domain/current_domain_api_external_experimental.h

```diff
@@ -78,13 +78,15 @@ TILEDB_EXPORT capi_return_t tiledb_current_domain_create(
  * @code{.c}
  * tiledb_current_domain_t *current_domain;
  * tiledb_current_domain_create(ctx, &current_domain);
- * tiledb_current_domain_free(&current_domain);
+ * tiledb_current_domain_free(ctx, &current_domain);
  * @endcode
  *
+ * @param ctx The TileDB context
  * @param current_domain The current domain to be freed
  * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
  */
 TILEDB_EXPORT capi_return_t tiledb_current_domain_free(
+    tiledb_ctx_t* ctx,
```

Comment:
```suggestion
```

Reply:
@teo-tsirpanis allocator failure would be the third option.
This was mostly a %s// job, given that it'll burden your C# work and that the generic capi handle error message is enough for the frequency I expect `free()` calls to fail here, I'm ok getting rid of the ctx arg.

Reply:
Are allocators expected to fail when freeing memory? I can imagine resource cleanup would be very messy if they did, akin to destructors not being allowed to throw. `free` does not return an error code for example.

Reply:
Opened SC-52107 to track removing the context parameter (or make it optional) from the other `free` APIs.